crates/ty_python_semantic/src/types.rs

```diff
@@ -4684,6 +4659,47 @@ impl<'db> Type<'db> {
         }
     }
 
+    /// Try to call `__getattr__` on the given type, respecting restrictions for fake typeshed methods.
+    ///
+    /// This helper function encapsulates the logic for calling `__getattr__` while preventing
+    /// the fake `__getattr__` methods on `types.ModuleType` and `types.GenericAlias` from being used.
+    fn try_custom_getattr(
```

Comment:
I think David's suggestion to extract this method was based on the theory that we'd reuse it for module-getattr, which we don't do in this PR. Given that we don't reuse it, I would probably revert the extraction; I don't think we need it as a full method if it's still just used in one place.

(At the very least we should remove the `allow_module_type_getattr` flag, which only ever receives the value `false`.)